code/game/machinery/doors/airlock.dm

```diff
@@ -1131,8 +1131,8 @@
 
 /obj/machinery/door/airlock/try_to_weld(obj/item/weapon/weldingtool/W, mob/user)
 	if(!operating && density)
-		if(W.remove_fuel(0,user))
-			if(user.a_intent != INTENT_HELP)
+		if(user.a_intent != INTENT_HELP)
+			if(W.remove_fuel(0,user))
```

Comment:
The welders could probably use a sort of direction, right now it's kinda random on how the fuel works, since sometimes it will remove fuel on use(while also removing fuel while it's on). But this can be done after the freeze.